data/binding/printf.go

```diff
@@ -0,0 +1,250 @@
+package binding
+
+import (
+	"fmt"
+)
+
+type sprintfString struct {
+	String
+
+	format string
+	source []DataItem
+	err    error
+}
+
+// NewSprintf returns a String binding that format its content using the
+// format string and the provide additional parameter that must be other
+// data bindings.
+//
+// Since: 2.2
+func NewSprintf(format string, b ...DataItem) (String, error) {
+	ret := &sprintfString{String: NewString(), format: format}
+
+	for _, value := range b {
+		value.AddListener(ret)
+
+		ret.source = append(ret.source, value)
+	}
+
+	return ret, nil
+}
+
+func (s *sprintfString) DataChanged() {
+	data := make([]interface{}, 0)
+
+	s.err = nil
+	for _, value := range s.source {
+		switch x := value.(type) {
+		case Bool:
+			{
+				b, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, b)
+				break
+			}
+		case Bytes:
+			{
+				b, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, b)
+				break
+			}
+		case Float:
+			{
+				f, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, f)
+				break
+			}
+		case Int:
+			{
+				i, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, i)
+				break
+			}
+		case Rune:
+			{
+				r, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, r)
+				break
+			}
+		case String:
+			{
+				str, err := x.Get()
+				if err != nil {
+					s.err = err
+					// Set error?
+					return
+				}
+
+				data = append(data, str)
+				break
+			}
+		case URI:
+			{
+				u, err := x.Get()
+				if err != nil {
+					s.err = err
+					return
+				}
+
+				data = append(data, u)
+				break
+			}
+		}
+	}
+
+	r := fmt.Sprintf(s.format, data...)
+	s.String.Set(r)
+}
+
+func (s *sprintfString) Get() (string, error) {
+	if s.err != nil {
+		return "", s.err
+	}
+	return s.String.Get()
+}
+
+func (s *sprintfString) Set(str string) error {
+	data := make([]interface{}, 0)
```

Comment:
Any reason to not just do `data := []interface{}` (applies to more lines)?

Reply:
Not knowing about this syntax :-) Will correct.

Reply:
It actually doesn't compile if I use the syntax you are suggesting.

Reply:
That's strange. Might be something specific to empty interfaces then?
It certainly does work for other types https://www.geeksforgeeks.org/slices-in-golang/.

Reply:
Try `var data []interface{}`

Reply:
You are right. It should work with `var data []interface{}` though.